File: LeetCode/Tag/Hash Table/java/1679.max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        int count = 0;
        for (int x : nums) {
            int minFreq = 0;
            if (freq.getOrDefault(k-x, 0) > 0) {
                ++count;
                freq.put(k-x, freq.get(k-x) - 1);
            } else {
                freq.put(x, freq.getOrDefault(x, 0) + 1);
            }
        }
        return count;
    }
}

File: LeetCode/Tag/Two-Pointers/cpp/1679.max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int count = 0, sum;
        for (int i = 0, j = nums.size()-1; i < j; ) {
            sum = nums[i] + nums[j];
            if (sum == k) { ++count; ++i; --j; }
            else if (sum < k) ++i;
            else --j;
        }
        return count;
    }
};